app/models/library_entry.rb

```diff
@@ -164,13 +167,26 @@ def kind
   end
 
   before_save do
-    if status_changed? && completed? && media&.progress_limit
-      # When marked completed, we try to update progress to the cap
-      self.progress = media.progress_limit
-    elsif !status_changed? && progress == media&.progress_limit
+    if status_changed?
```

Comment:
please... no.... :sob:

Nested if's are generally a big nono, nested nested if's are.... inhumane.... It might be best to break this out into a few methods.